src/clang.rs

```diff
@@ -647,11 +647,17 @@ impl Type {
 
     /// Given that this type is a pointer type, return the type that it points
     /// to.
-    pub fn pointee_type(&self) -> Type {
-        unsafe {
-            Type {
-                x: clang_getPointeeType(self.x),
+    pub fn pointee_type(&self) -> Option<Type> {
+        match self.x.kind {
+            CXType_Pointer |
+            CXType_RValueReference |
+            CXType_LValueReference |
+            CXType_MemberPointer => {
+                unsafe {
+                    Some(Type { x: clang_getPointeeType(self.x) })
+                }
```

Comment:
Huh, for some reason my other comment was lost.

Can you assert the type is valid here?

```rust
let ret = Type {
    x: unsafe { clang_getPointeeType(self.x) },
};
debug_assert!(ret.kind() != CXType_Invalid);
Some(ret)